.github/workflows/checks.yaml

```diff
@@ -392,3 +392,96 @@ jobs:
         run: "cc test-libbzip2-rs-sys/no_stdio.c -o no_stdio target/release/liblibbzip2_rs_sys.a -I ."
       - name: run C binary
         run: "./no_stdio | grep 'hello world!'"
+
+  link-c-dynamic-library:
+    name: vanilla dynamic library
+    strategy:
+      matrix:
+        include:
+          - target: x86_64-unknown-linux-gnu
+        features:
+          - ''
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+        with:
+          persist-credentials: false
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@be73d7920c329f220ce78e0234b8f96b7ae60248
+        with:
+          toolchain: stable
+          targets: ${{matrix.target}}
+      - name: "cdylib: default settings"
+        working-directory: libbzip2-rs-sys-cdylib
+        env:
+          LD_LIBRARY_PATH: "target/${{matrix.target}}/release/deps"
+        run: |
+          cargo build --release --target ${{matrix.target}}
+          cc -o bzpipe bzpipe.c target/${{matrix.target}}/release/deps/libbz2_rs.so -I ../
+          ./bzpipe < Cargo.toml | ./bzpipe -d > out.txt
+          cmp -s Cargo.toml out.txt
+      - name: "cdylib: rust-allocator"
+        env:
+          LD_LIBRARY_PATH: "target/${{matrix.target}}/release/deps"
+        working-directory: libbzip2-rs-sys-cdylib
+        run: |
+          cargo build --release --target ${{matrix.target}} --no-default-features --features="rust-allocator"
+          cc -o bzpipe bzpipe.c target/${{matrix.target}}/release/deps/libbz2_rs.so -I ../
+          ./bzpipe < Cargo.toml | ./bzpipe -d > out.txt
+          cmp -s Cargo.toml out.txt
+      #      - name: "cdylib: no_std"
+      #        env:
+      #          LD_LIBRARY_PATH: "target/${{matrix.target}}/release/deps"
+      #        working-directory: libbzip2-rs-sys-cdylib
+      #        run: |
+      #          cargo build --release --target ${{matrix.target}} --no-default-features
+      #          cc -o bzpipe_no_std bzpipe_no_std.c target/${{matrix.target}}/release/deps/libbz2_rs.so -I ../
+      #          ./bzpipe_no_std < Cargo.toml | ./bzpipe_no_std -d > out.txt
+      #          cmp -s Cargo.toml out.txt
```

Comment:
I'm still not sure how this works for zlib-rs and doesn't here

Reply:
got it, the `[lib]` section of the main crate needs to be removed; that's allright, it is only meant to be used as a crate anyway, building libraries should happen with the cdylib crate.